### PR TITLE
SC2: Add Archon upgrades

### PR DESCRIPTION
**Pull request by Snarkie**
## What is this fixing or adding?
Adds 4 upgrade items to Archons
- Transcendence: Archons float, can traverse cliffs and phase through most units
- Power Siphon: Autocast single target ability, that deals damage over time and restores shields per damage dealt
- Obliterate: Upgrades main weapon with increased AoE, bonus damage to biological is new base damage
- Eradicate: Archons launch themselves to nearby ground units on death, dealing heavy damage in a small area

## How was this tested?
Generated a local campaign, added the items with commands. Tested interactions with other items

[Client PR](https://github.com/Ziktofel/Archipelago/pull/368)

### CODE REVIEW

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml [hunk 1]
```diff
@@ -16915,4 +16926,31 @@
         <EffectArray Operation="Set" Reference="Button,AP_BanelingLaunch,Tooltip" Value="Button/Tooltip/AP_BanelingLaunchLevel2"/>
         <EffectArray Operation="Set" Reference="Abil,AP_BanelingLaunchHangar,InfoArray[Ammo1].Button.Requirements" Value="AP_BanelingLaunchArmLevel2"/>
     </CUpgrade>
+    <CUpgrade id="AP_ArchonFloat">
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Mover" Value="Colossus"/>
+        <!-- 1 = ground, 8 = forcefield, 15 = small, 4 = colossus, 5 = structure -->
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[1]" Value="0"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[8]" Value="0"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[15]" Value="0"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_Archon,Collide[4]" Value="1"/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml:16935]: Unupgraded don't collide with colossi, the upgrade could bring you into problems with using such units (a possible downgrade)

**Review by Ziktofel — CHANGES_REQUESTED** [review]
Snímek obrazovky pořízený 2024-12-10 18-58-08
Missing Range Actor

Snímek obrazovky pořízený 2024-12-10 19-05-59
Missing related model

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml [hunk 2]
```diff
@@ -23127,4 +23127,107 @@
...
+        <InitialEffect value="AP_ArchonSiphon"/>
+        <PeriodCount value="15"/>
+        <PeriodicValidator value="AP_TargetNotDeadAndCasterNotDead"/>
+        <PeriodicEffectArray value="AP_ArchonSiphonSet"/>
+        <PeriodicPeriodArray value="0.125"/>
+    </CEffectCreatePersistent>
+    <CEffectDamage id="AP_ArchonSuicideDamage">
+        <EditorCategories value=""/>
+        <AttributeBonus index="Armored" value="50"/>
+        <AttributeBonus index="Massive" value="125"/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml:23171]: Doesn't work as intended: Thors will receive 162 damage (if multiple attribute bonuses happen, you get average the damage). Sky Furies have a workaround there with multiple damage effects so they stack properly
You need some better dmg handling there

What's the intended dmg to:
- Armored but not Massive?
- Massive but not Armored?
- Both Massive and Armored

File: Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt [hunk 3]
```diff
@@ -2218,7 +2222,11 @@ Button/Tooltip/AP_ArbiterMPRecall=Teleports all friendly units in the target are
...
 Button/Tooltip/AP_ArbiterReinforcedHull=Increases Arbiter life by 50 and armor by 1.
 Button/Tooltip/AP_Archon=Powerful psionic attacker.<n/><n/><c val="#ColorAttackInfo">Can attack ground and air units.</c>
+Button/Tooltip/AP_ArchonFloat=Archons float in the air. Allows Archons to traverse cliffs and phase through most units.
 Button/Tooltip/AP_ArchonHallucination=These hallucinated copies function like the original unit, but they cannot use active abilities, deal damage, warp in structures, or gather resources, and they die more easily. Hallucinations last <d time="60"/> before they disappear.
+Button/Tooltip/AP_ArchonSiphon=Deals 80 damage over 2 seconds to the target unit, and recovers 2 Shields per damage dealt.<n/><n/>Autocasts when at 50% shield or lower.<n/><c val="#ColorAttackInfo">Can target ground and air units.</c>
+Button/Tooltip/AP_ArchonSuicide=The Archon launches at a nearby enemy on death, dealing 75 damage in a small area. (125 against armored, 200 against massive).<n/><n/><c val="#ColorAttackInfo">Can target ground units.</c>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt:2228]: Ground units only

**Review by Ziktofel — CHANGES_REQUESTED** [review]

**Comment by Ziktofel** [comment]
Check all your linked assets if they exist in `liberty.sc2mod`

File: Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameHotkeys.txt [hunk 4]
```diff
@@ -604,3 +604,4 @@ Button/Hotkey/AP_YggdrasilAssimilate=T
 Button/Hotkey/AP_SILiberatorAAMode=D
 Button/Hotkey/AP_SILiberatorAGMode=E
 Button/Hotkey/AP_SummonMercRoach=T
+Button/Hotkey/AP_ArchonSiphon=Q
```
[the remark this answers is not in the document]
Comment (Snarkie) [reply Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameHotkeys.txt]: Changed to R

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml (hunk 1 above)
Comment (Snarkie) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml:16935]: Changed so floating Archons get collision flag 12, which is only shared with Reaper charges. Best ground collision flag I could find, so they still block each other.

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml (hunk 2 above)
Comment (Snarkie) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml]: Changed to 75 + 75 to massive. The different bonuses to armored and massive were somewhat random. Easier to just change it, will probably be rebalanced in polishing stage anyway.

File: Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt (hunk 3 above)
Comment (Snarkie) [reply Mods/ArchipelagoPlayer.SC2Mod/enUS.SC2Data/LocalizedData/GameStrings.txt]: fixed

**Comment by Snarkie** [comment]
> Check all your linked assets if they exist in `liberty.sc2mod`

all missing assets should be fixed.

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/SoundData.xml [hunk 5]
```diff
@@ -14952,4 +14952,17 @@
         <VolumeRolloffPoints Distance="20.353981" Volume="-96.000000"/>
         <Pitch value="-2.000000,0.000000"/>
     </CSound>
+    <CSound id="AP_ArchonSiphon">
+        <AssetArray File="Assets\Sounds\Protoss\VoidSeeker\VoidSeeker_PsytrousOxideMovement1.wav" FacialGroup=""/>
+        <DupeDestroyCount value="2"/>
+    </CSound>
+    <CSound id="AP_Archon_AttackLaunchUpgrade" parent="Combat">
+        <EditorCategories value="Race:Protoss"/>
+        <AssetArray File="AP\Assets\Sounds\Protoss\ArchonSCBW\ArchonSCBWWeapon@Launch00.ogg" FacialGroup=""/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/SoundData.xml:14961]: The sound is too loud

**Pull request merged**